frontend/src/components/DrawerDetails/ResourceDetailsDrawer.tsx

```diff
@@ -0,0 +1,152 @@
+import React, { useEffect, useState } from "react";
+import { Drawer, Spin, Collapse, Typography, Tooltip } from "antd";
+import { HelmDataSourceItem, ResourceDataSourceItem } from "../../types";
+import { getResource } from "../../api/k8s/getResource";
+import styles from "./ResourceDetailsDrawer.module.css";
+import {fetchRelease, fetchReleaseHistory} from "../../api";
+
+const { Paragraph } = Typography;
+const { Panel } = Collapse;
+
+interface DrawerDetailsProps {
+    visible: boolean;
+    record: ResourceDataSourceItem | HelmDataSourceItem | object | null;
+    onClose: () => void;
+    loading: boolean;
+    resourceType: string;
+}
+
+const ResourceDetailsDrawer: React.FC<DrawerDetailsProps> = ({
+                                                                 visible,
+                                                                 record,
+                                                                 onClose,
+                                                                 resourceType,
+                                                             }) => {
+    const [resourceDetails, setResourceDetails] = useState<any | null>(null);
+    const [fetching, setFetching] = useState(false);
+
+    const fetchResourceData = async (
+        resourceType: string,
+        resourceName: string,
+        namespace: string
+    ) => {
+        setFetching(true);
+        try {
+            if (resourceType !== "Helm") {
+                const details = await getResource(resourceType, resourceName, namespace);
+                setResourceDetails(details);
+                console.log(details);
+            } else {
+                const details = await fetchRelease(resourceName, namespace);
+                const history = await fetchReleaseHistory(resourceName, namespace);
+
+                const combinedDetails = [details, history];
+
+                setResourceDetails(combinedDetails);
+                console.log(combinedDetails);
+
+            }
+        } catch (error: unknown) {
+            if (error instanceof Error) {
+                console.error("Error fetching resource details:", error.message);
+            } else {
+                console.error("Unknown error:", error);
```

Comment:
It would be good to add messages.

Reply:
For what? The info about error fetching is in the drawer.

Reply:
You are right.